src/graph/graph.c

```diff
@@ -812,7 +811,12 @@ void Graph_DeleteNode
 	uint32_t label_count = array_len(g->labels);
 	for(int i = 0; i < label_count; i++) {
 		RG_Matrix M = Graph_GetLabelMatrix(g, i);
-		RG_Matrix_removeElement_BOOL(M, ENTITY_GET_ID(n), ENTITY_GET_ID(n));
+		GrB_Info res = RG_Matrix_removeElement_BOOL(M, ENTITY_GET_ID(n),
+				ENTITY_GET_ID(n));
+		if(res == GrB_SUCCESS) {
+			// a node with this label has just been deleted, update statistics
+			GraphStatistics_DecNodeCount(&g->stats, i, 1);
+		}
 	}
```

Comment:
It seems to be more efficient to use `int label_id = NODE_GET_LABEL_ID(n, g);` here instead of always iterating,
there are some cases like `op_node_by_label_scan` where we set the node's label.
Please add a TODO here to revisit when `MULTI-LABEL` is merged.